### PR TITLE
New versions of Zircon and Amethyst

### LINKED ISSUE #124: Update Zircon
Zircon will come out with a new major version soon, but there is already `2020.1.9-PREVIEW` to be tested. We should update to that version and try it. We can then see what breaks and what needs to be fixed and if the game still works. This is valuable feedback for Hexworks.

### PR DESCRIPTION
**Pull request by Baret**
Resolves #124

### CODE REVIEW

**Comment by Baret** [comment]
Zircon and Cobalt are updated. The parts where `MultiSelect` changed to `Selector` are much simpler now. And as a small goodie I added a 3D-Toggle to the map generation view :)

I forgot Amethyst for now but have a review for the zircon changes. I will do Amethyst afterwards. Or maybe in a new issue...? It might come with some cool changes we will want to implement.

**Review by Loomie — COMMENTED** [review]
Looks not too bad ;) The 3D view is great!

File: pom.xml [hunk 1]
```diff
@@ -34,9 +34,9 @@
 		<!-- version used by Kotest -->
 		<kotlinx.coroutines.version>1.3.9</kotlinx.coroutines.version>
 
-		<zircon.version>2020.1.5-PREVIEW</zircon.version>
+		<zircon.version>2020.1.9-PREVIEW</zircon.version>
```
Comment (Loomie) [inline pom.xml:37]: Uhm, why is this a PREVIEW version? https://hexworks.org/posts/news/2020/12/03/new-zircon-release-2020.2.0-RELEASE.html

File: game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/mapgeneration/IncompleteMapGameArea.kt [hunk 2]
```diff
@@ -4,53 +4,89 @@ import de.gleex.pltcmd.model.world.coordinate.MainCoordinate
...
+        val terrainZLevel = if (terrainHeight != null) {
+                TerrainHeight.values()
+                    .indexOf(terrainHeight)
+            } else {
+                Z_LEVEL_DEFAULT
             }
-            .toPersistentMap()
+        List(BLOCKS_HEIGHT) { toPosition3D(it) }
+                .forEach { position3D ->
+                    fetchBlockAt(position3D)
```
[the remark this answers is not in the document]
Comment (Baret) [reply game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/mapgeneration/IncompleteMapGameArea.kt:51]: Finally, welcome to Kotlin! 😁

File: game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/components/MapComponent.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * The main component of the game view. It displays the map with its decorations.
+ */
+class MapComponent(gameWorld: GameWorld) : Fragment {
+
+    private val mapRenderer = GameComponents.newGameAreaComponentRenderer<Panel, Tile, GameBlock>(gameWorld)
+
+    override val root: Panel = Components.panel()
+        .withSize(UiOptions.MAP_VIEW_WIDTH, UiOptions.MAP_VIEW_HEIGHT)
```
[the remark this answers is not in the document]
Comment (Baret) [reply game/ui/src/main/kotlin/de/gleex/pltcmd/game/ui/components/MapComponent.kt]: As we do not (yet) have stuff like layout managers or dynamic sizing/resizing we have to define window and component sizes statically...

File: pom.xml (hunk 1 above)
Comment (Baret) [reply pom.xml]: I started the branch just before the release came out to possibly provide feedback to [Hexworks](https://github.com/Hexworks/).

**Review by Baret — COMMENTED** [review]
Looks good to me! When you approve this PR go ahead and merge it @Loomie 👍

**Pull request merged**